MobileCenterUpdates/MobileCenterUpdates/MSUpdates.m

```diff
@@ -306,6 +317,87 @@ - (void)openURLInSafariApp:(NSURL *)url {
 }
 
 - (void)handleUpdate:(MSReleaseDetails *)details {
+
+  // Step 1. Validate release details.
+  if (![details isValid]) {
+    MSLogError([MSUpdates logTag], @"Received invalid release details.");
+    return;
+  }
+
+  // Step 2. Check status of the release. TODO: This will be deprecated soon.
+  if (![details.status isEqualToString:@"available"]) {
+    MSLogError([MSUpdates logTag], @"The new release is not available, skip update.");
+    return;
+  }
+
+  // Step 3. Check if the release ID was ignored by a user.
+  NSString *releaseId = [[MSUserDefaults shared] objectForKey:kMSIgnoredReleaseIdKey];
+  if (releaseId && [releaseId isEqualToString:details.id]) {
+    MSLogDebug([MSUpdates logTag], @"A user already ignored updating this release, skip update.");
+    return;
+  }
+
+  // Step 4. Check min OS version.
+  if ([MS_DEVICE.systemVersion compare:details.minOs options:NSNumericSearch] != NSOrderedAscending) {
```

Comment:
Are we sure that this is working with version numbers like 9.3.1 for instance? I couldn't have it working.